app/controllers/register_controller.rb

```diff
@@ -74,7 +72,7 @@ def confirm
   end
 
   def create
-    fields = get_register(params[:register_id]).fields
+    fields = @registers_client.get_register(params[:register_id], 'beta').get_register_definition.item.value['fields']
```

Comment:
Is this supposed to be hardcoded to beta? Just wondering because above we use `Rails.configuration.register_phase`